linera-core/src/unit_tests/client_tests.rs

```diff
@@ -2456,8 +2458,31 @@ where
     assert_eq!(executed_block.block.incoming_bundles.len(), 1);
     assert_eq!(executed_block.required_blob_ids().len(), 1);
 
+    // This will go way over the limit, because of the different overheads.
+    let blob_bytes = (0..100)
+        .map(|_| {
+            rand::thread_rng()
+                .sample_iter(&rand::distributions::Standard)
+                .take(policy.maximum_blob_size as usize)
+                .collect::<Vec<_>>()
+        })
+        .collect::<Vec<_>>();
+    let result = client1.publish_data_blobs(blob_bytes).await;
+    assert_matches!(
+        result,
+        Err(ChainClientError::ChainError(
+            ChainError::BlockProposalTooLarge
+        ))
+    );
+    client1.clear_pending_block();
```

Comment:
Good catch! Let's make the client clear that automatically if it fails before sending it to the validators.

Reply:
Will do! Issue: https://github.com/linera-io/linera-protocol/issues/2906